zboxcore/sdk/upload_worker.go

```diff
@@ -0,0 +1,146 @@
+package sdk
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	thrown "github.com/0chain/errors"
+	"github.com/0chain/gosdk/constants"
+	"github.com/0chain/gosdk/zboxcore/allocationchange"
+	"github.com/0chain/gosdk/zboxcore/fileref"
+	l "github.com/0chain/gosdk/zboxcore/logger"
+	"github.com/0chain/gosdk/zboxcore/zboxutil"
+	"github.com/google/uuid"
+)
+
+type UploadOperation struct {
+	workdir    string
+	fileMeta   FileMeta
+	fileReader io.Reader
+	opts       []ChunkedUploadOption
+	refs       []fileref.FileRef
+	isUpdate   bool
+	statusCallback StatusCallback
+	opCode int
+}
+
+func (uo *UploadOperation) Process(allocObj *Allocation, connectionID string, totalOperation int) ([]fileref.RefEntity, error) {
+	cu, err := CreateChunkedUpload(uo.workdir, allocObj, uo.fileMeta, uo.fileReader, uo.isUpdate, false, connectionID, uo.opts...)
+	uo.statusCallback = cu.statusCallback;
+	uo.opCode = cu.opCode;
+	if err != nil {
+		return nil, err
+	}
+	cu.uploadTimeOut = DefaultUploadTimeOut * time.Duration(totalOperation); 
+	err = cu.process()
+	if err != nil {
+		return nil, err
+	}
+
+	var pos uint64
+	numList := len(cu.blobbers)
+	objectTreeRefsEntity := make([]fileref.RefEntity, numList)
+	uo.refs = make([]fileref.FileRef, numList)
+	for i := cu.uploadMask; !i.Equals64(0); i = i.And(zboxutil.NewUint128(1).Lsh(pos).Not()) {
+		pos = uint64(i.TrailingZeros())
+		objectTreeRefsEntity[pos] = cu.blobbers[pos].fileRef
+		uo.refs[pos] = *cu.blobbers[pos].fileRef
+		uo.refs[pos].NumBlocks = int64(cu.progress.ChunkIndex + 1)
+		uo.refs[pos].ChunkSize = cu.chunkSize
+	}
+
+	l.Logger.Info("Completed the upload")
+	return objectTreeRefsEntity, nil
```

Comment:
If we don't use it in buildChange then send nil. Or send actual value and use it in buildChange as well.